multimodal/vl2l/src/mlperf_inference_multimodal_vl2l/task.py

```diff
@@ -230,7 +236,88 @@ async def _query_endpoint_async(
                             query_sample.id,
                             address,
                             size_in_bytes,
-                            len(content),
+                            int(response.usage.completion_tokens),
+                        ),
+                    ],
+                )
+
+            for query_sample in query_samples:
+                asyncio.run_coroutine_threadsafe(
+                    _query_endpoint_async(query_sample),
+                    self.event_loop,
+                )
+
+        def _issue_streaming_queries(
+                query_samples: list[lg.QuerySample]) -> None:
+            """Called by the LoadGen to issue queries to the inference endpoint.
+
+            Args:
+                query_samples: The list of query samples to issue to the inference
+                    endpoint. Each query sample contains (1) `id`: `lg.ResponseId`
+                    (i.e., unique identifier for the response), and (2) `index`:
+                    `lg.QuerySampleIndex` (i.e., the sample index into the dataset).
+            """
+
+            async def _query_endpoint_async(
+                    query_sample: lg.QuerySample) -> None:
+                """Query the endpoint through the async OpenAI API client."""
+                messages = self.loaded_messages[query_sample.index]
+                logger.trace(
+                    "Issuing query sample index: {} with response ID: {}",
+                    query_sample.index,
+                    query_sample.id,
+                )
+                ttft_set = False
+                word_array = []
+                stream = await self.openai_api_client.chat.completions.create(
+                    stream=True,
+                    model=self.model_cli.repo_id,
+                    messages=messages,
+                    stream_options={"include_usage": True},
+                )
+                # iterate asynchronously
+                total_tokens = 0
+                async for chunk in stream:
+
+                    # This is the final chunk and will not have 'choices'
+                    if chunk.usage is not None:
+                        total_tokens = int(chunk.usage.completion_tokens)
+
+                    # If it's not the usage chunk, process it as a content
+                    # chunk
+                    choices = getattr(chunk, "choices", None)
+                    if not choices:
+                        continue
+                    # first non-empty token -> TTFT
+                    delta = chunk.choices[0].delta
```

Comment:
```suggestion
                    delta = choices[0].delta
```
You already took out `choices` in the line above.